File: leetcode/data_struct/string/0058/1.go
// auther: phoenix
// date: 2023-06-13 07:39:19
package main

import "strings"

/*


 */

func main() {
	lengthOfLastWord1Test()
}

func lengthOfLastWord1Test() {
	println(lengthOfLastWord1("Hello World"))
	println(lengthOfLastWord1("   fly me   to   the moon  "))
	println(lengthOfLastWord1("luffy is still joyboy"))
}

func lengthOfLastWord1(s string) int {
	str := strings.Trim(s, " ")
	strs := strings.Split(str, " ")
	lastWord := strs[len(strs)-1]

	return len(lastWord)
}

File: leetcode/data_struct/string/0058/1.rs
// auther: phoenix
// date: 2023-06-13 07:39:24

/*



*/

fn main() {
  length_of_last_word1_test();
}

fn length_of_last_word1_test () {
  println!("{}", length_of_last_word1(String::from("Hello World")));
  println!("{}", length_of_last_word1(String::from("   fly me   to   the moon  ")));
  println!("{}", length_of_last_word1(String::from("luffy is still joyboy")));
}
fn length_of_last_word1(s: String) -> usize {
  let s1:&str = s.trim();
  let strs:Vec<&str> = s1.split(' ').collect();
  let last_word:&str = strs[strs.len() - 1];
  
  last_word.len()
}